src/interceptors/ClientRequest/utils/normalizeHttpRequestParams.ts

```diff
@@ -52,15 +55,21 @@ export function normalizeHttpRequestParams(
     debug('created request options:', options)
 
     callback = resolveCallback(args)
-  } else if ('origin' in args[0]) {
+  }
+  // Handle a given URL instance as-is
+  // and derive request options from it.
+  else if ('origin' in args[0]) {
     url = args[0]
     debug('given a URL:', url)
 
     options = resolveRequestOptions(args, url)
     debug('created request options', options)
 
     callback = resolveCallback(args)
-  } else if ('method' in args[0]) {
+  }
+  // Handle a given request options object as-is
+  // and derive the URL instance from it.
+  else if (isObject(args[0])) {
```

Comment:
I've isolated this predicate into its own utility, making this code easier to read.
I've also added a brief explanatory comment above each `if` statement, so it could be quicker to understand what this case handles.